### PR TITLE
refactor builders to share tar and gzip code

### PR DESCRIPTION
**Pull request by aaron-prindle**

### CODE REVIEW

**Comment by dlorenc** [comment]
Some flake/formatting issues.

File: ftl/common/ftl_util.py [hunk 1]
```diff
@@ -27,3 +31,18 @@ def __enter__(self):
     def __exit__(self, unused_type, unused_value, unused_traceback):
         end = time.time()
         logging.info('%s took %d seconds', self.descriptor, end - self.start)
+
+
+def folder_to_layer_sha(pkg_dir, pkg_mngr_name):
```
Comment (dlorenc) [inline ftl/common/ftl_util.py]: Maybe "zip_directory" would be more clear?

File: ftl/common/ftl_util.py [hunk 2]
```diff
@@ -27,3 +31,18 @@ def __enter__(self):
...
+    tar_path = tempfile.mktemp()
+    with Timing("tar_%s_package" % pkg_mngr_name):
```
Comment (dlorenc) [inline ftl/common/ftl_util.py]: It's a bit weird we need the package manager name here - there's nothing else that uses this except the logging statement. Can we remove that as a parameter?

File: ftl/common/ftl_util.py (hunk 1 above)
Comment (aaron-prindle) [reply ftl/common/ftl_util.py]: "zip_directory_to_layer_sha"?  This takes a path to a folder that is uncompressed/untarred

File: ftl/common/ftl_util.py (hunk 2 above)
Comment (aaron-prindle) [reply ftl/common/ftl_util.py]: done

File: ftl/common/ftl_util.py (hunk 1 above)
Comment (aaron-prindle) [reply ftl/common/ftl_util.py]: done

**Comment by aaron-prindle** [comment]
also wanted to mention this is not every gzip call.  A similar thing is done in ftl/common/builder.py:
https://github.com/GoogleCloudPlatform/runtimes-common/blob/master/ftl/common/builder.py#L68-L87
It is different though in that it is retrieving files from a context that is not necessarily on disk.  For the actual FTL pipeline it might make sense to have this use the same method but we have MemoryContexts that wouldn't work a subprocess.check_call('tar', '...')

**Comment by dlorenc** [comment]
> It is different though in that it is retrieving files from a context that is not necessarily on disk. For the actual FTL pipeline it might make sense to have this use the same method but we have MemoryContexts that wouldn't work a subprocess.check_call('tar', '...')

Ah, yeah. Maybe it would make sense to give the context interface a method to "tar/zip" itself up, similar to this. Then the disk context could take advantage of this optimization.

**Comment by aaron-prindle** [comment]
I'll change this to be a part of the context interface

**Pull request merged**